src/core/provider/MCAccountValidator.h

```diff
@@ -0,0 +1,96 @@
+//
+//  MCAccountValidator.h
+//  mailcore2
+//
+//  Created by Christopher Hockley on 22/01/15.
+//  Copyright (c) 2015 MailCore. All rights reserved.
+//
+
+#ifndef MAILCORE_MCACCOUNTVALIDATOR_H
+
+#define MAILCORE_MCACCOUNTVALIDATOR_H
+
+#include <MailCore/MCBaseTypes.h>
+
+#ifdef __cplusplus
+
+namespace mailcore {
+    
+    class NetService;
+    class MailProvider;
+    class IMAPOperation;
+    
+    class MAILCORE_EXPORT AccountValidator : public Operation, public OperationCallback {
+    public:
+        AccountValidator();
+        virtual ~AccountValidator();
+        
+        virtual void setEmail(String * email);
+        virtual String * email(); /* for SMTP */
+        virtual void setUsername(String * username);
+        virtual String * username();
+        virtual void setPassword(String * password);
+        virtual String * password();
+        virtual void setOAuth2Token(String * OAuth2Token);
+        virtual String * OAuth2Token();
+        
+        virtual void setImapServices(Array * imapServices);
+        virtual Array * /* NetService */ imapServices();
+        virtual void setSmtpServices(Array * smtpServices);
+        virtual Array * /* NetService */ smtpServices();
+        virtual void setPopServices(Array * popServices);
+        virtual Array * /* NetService */ popServices();
+        
+        // result
+        virtual String * identifier();
+        virtual NetService * imapServer();
+        virtual NetService * popServer();
+        virtual NetService * smtpServer();
+        virtual ErrorCode imapError();
+        virtual ErrorCode popError();
+        virtual ErrorCode smtpError();
+        
+        virtual void start();
```

Comment:
Isn't the cancel() from Operation called? 
In which case I should still add something like 
if (isCancelled())
        return;
At the start of my operationFinished method?